src/components/page/AppPage.js

```diff
@@ -83,55 +85,59 @@ class AppPage extends React.Component {
         super();
 
         this.state = {
-            imgList: [null, null],
-            modelReady: false,
-            gauging: false,
+            descriptors: [null, null],
+            status: Status.NONE,
             similarity: 0,
         };
+    }
 
-        this.loadModel().then(_ => {
-            this.setState({ modelReady: true });
-        });
+    componentDidMount() {
+        this.loadModel().catch(console.error);
     }
 
     loadModel = async () => {
+        this.setState({ status: Status.LOADING_MODEL });
         await faceApi.loadFaceRecognitionModel('/models');
+        await faceApi.loadSsdMobilenetv1Model('/models');
+        await faceApi.loadFaceLandmarkModel('/models');
+        this.setState({ status: Status.NONE });
     };
 
     onClickGauge = async () => {
-        this.setState({ gauging: true });
-        const inputs = [undefined, undefined];
-        const descriptors = [undefined, undefined];
-
-        for (let i = 0; i < 2; i++) {
-            inputs[i] = await faceApi.fetchImage(this.state.imgList[i]);
-            descriptors[i] = await faceApi.computeFaceDescriptor(inputs[i]);
+        if (!this.state.descriptors[0] || !this.state.descriptors[1]) {
+            alert('Upload 2 face images!');
```

Comment:
What about msging 'Please upload each of two face images!' ?

Reply:
Thanks ! Done